sklearn/titanic/README.md

```diff
@@ -0,0 +1,125 @@
+```bash
+# Run on the new instance
+sudo apt-get install -y python-pip
+pip install numpy pandas sklearn scipy tensorflow
+```
+Note: If you choose a different image for the instance, you may need to do something different to install pip and the required python packages.
+
+Please note that we are not using *tensorflow* to train the model in this sample. We will however use *gfile*, a python class implemented in *tensorflow* which unifies how we access the local and remote files. An alternative to using *gfile*  for accessing the files stored in *Google Cloud Storage (GCS)* is [Cloud Storage Client Libraries](https://cloud.google.com/storage/docs/reference/libraries#client-libraries-install-python).
+
+If we want to use *GCS* to store the dataset, the new instance will then require to obtain user access credentials. The following command will guide us through the process:
+```bash
+# Run on the new instance
+gcloud auth application-default login
+```
+
+##### 3. Running the Code
+We need to make the dataset available to the training code. Here are two ways for the code to access the dataset.
+
+###### Option 1: Copying it to the instance
+We can copy the dataset into the new instance and use it directly inside the instance and read it as a local file. Fortunately, *gcloud* makes this an easy step:
+```bash
+gcloud compute scp ./train.csv $MY_INSTANCE:~/
+```
+
+Note that the same method can be used to copy any other file (e.g. the python code):
+```bash
+gcloud compute scp ./titanic.py $MY_INSTANCE:~/
+```
+
+
+We can then run our python code in the instance to create a model:
+```bash
+# Run on the new instance
+python titanic.py --titanic-data-path ~/train.csv --model-output-path ~/model.pkl
+```
+
+which will generate the model and save it on the instance.
+
+###### Option 2: Reading it from GCS
+Another solution is to read the dataset directly from a bucket in GCS. We use *gsutil* to create the bucket in GCS, and then copy the dataset into it:
+```bash
+gsutil mb $MY_BUCKET
+gsutil cp ./train.csv  gs://$MY_BUCKET/$MY_FOLDER/train.csv
+```
+The advantage of using *gfile* in our python code is that we can use the exact same code and read the dataset from GCS:
+```bash
+# Run on the new instance
+export $MY_BUCKET=THE_GIVEN_BUCKET_NAME_HERE
+export $MY_FOLDER=THE_GIVEN_FOLDER_NAME_HERE
+python titanic.py --titanic-data-path gs://$MY_BUCKET/$MY_FOLDER/train.csv --model-output-path gs://$MY_BUCKET/$MY_FOLDER/model.pkl
+```
+
+which will create the model and store it in the same location as our dataset in *GCS*.
+
+That is it!! We have successfully trained a classifier using scikit-learn on a *GCE* instance.
+
+##### 4. Stopping or Deleting the Instance
+The *GCE* instance that we created will be running indefinitely and costing us, unless we either stop or delete it completely. Stopping the instance will allow us to start it again later on, should we require to use it again which may be the better choice:
```

Comment:
Little confused on this line. The questions that jump out when I read this. 
What's costing me? Does having a stopped but saved instance cost me money?

Reply:
I changed it to make it more clear.